rclcpp/src/rclcpp/utilities.cpp

```diff
@@ -117,7 +117,17 @@ rclcpp::utilities::init(int argc, char * argv[])
     // NOLINTNEXTLINE(runtime/arrays)
     char error_string[error_length];
 #ifndef _WIN32
-    strerror_r(errno, error_string, error_length);
+#ifdef _GNU_SOURCE
+    char * msg = strerror_r(errno, error_string, error_length);
+    if (msg != error_string) {
+      strncpy(error_string, msg, sizeof(error_string));
+    }
```

Comment:
If `msg` would be a string with 1024 or more characters and different from `error_string` the `strncpy` will make `error_string` a non-null terminated string which will like make the subsequent throw in line 135 explode.